index.js

```diff
@@ -102,6 +115,54 @@ function fastifyPostgres (fastify, options, next) {
     }
   }
 
+  fastify.addHook('onRoute', routeOptions => {
+    const useTransaction = routeOptions.useTransaction || (routeOptions.options && routeOptions.options.useTransaction)
+
+    if (useTransaction) {
+      // This will rollback the transaction if the handler fails at some point
+      const onError = async (req, reply, error) => {
+        req.transactionFailed = true
+
+        try {
+          await req.pg.query('ROLLBACK')
+        } catch (err) {
+          await req.pg.query('ROLLBACK')
+        }
```

Comment:
what does this do?

Reply:
This is the best I could come up with in the response to the question 'what happens if rollback fails'

Reply:
if rollback fails and you try it again, it will most likely fail again. so retrying a rollback is not useful. rather, we should make sure that failing a rollback due to an error happened in the handler returns an error to the client which reflects the error that was thrown in the handler, whatever it was

Reply:
Does Fastify already do that? ie. `onError` intercepts an error that was already heading to the user via `500` mechanisms?

https://www.fastify.io/docs/latest/Hooks/#onerror

The only option would seem to be throwing a 'rollback failed' 

Wondering what happens to unfinished transactions who's clients are released?